webviz_subsurface/_datainput/eclipse_init_io/pvt_water.py

```diff
@@ -0,0 +1,408 @@
+            self.original_unit_system
+            if self.keep_unit_system
+            else EclUnitEnum.ECL_SI_UNITS
+        )
+
+        if latex:
+            return (
+                fr"${{{unit_system.reservoir_volume().symbol}}}"
+                fr"/{{{unit_system.surface_volume_liquid().symbol}}}$"
+            )
+        return (
+            f"{unit_system.reservoir_volume().symbol}"
+            f"/{unit_system.surface_volume_liquid().symbol}"
+        )
+
+    def viscosity_unit(self, latex: bool = False) -> str:
+        """Creates and returns a string containing the unit symbol of the viscosity.
+
+        Args:
+            latex: True if the unit symbol shall be returned as LaTeX, False if not.
+
+        Returns:
+            A string containing the unit symbol of the viscosity.
+
+        """
+        unit_system = EclUnits.create_unit_system(
+            self.original_unit_system
+            if self.keep_unit_system
+            else EclUnitEnum.ECL_SI_UNITS
+        )
+
+        if latex:
+            return fr"${unit_system.viscosity().symbol}$"
+        return f"{unit_system.viscosity().symbol}"
+
+    def water_unit_converter(
+        self, unit_system: Union[int, EclUnits.UnitSystem]
+    ) -> ConvertUnits:
+        """Creates a ConvertUnits object for unit conversions for water.
+
+        Args:
+            unit_system:
+                Either an integer or an enum
+                describing the unit system the units are stored in
+
+        Returns:
+            ConvertUnits object for unit conversions.
+
+        """
+        if not isinstance(unit_system, EclUnits.UnitSystem):
+            unit_system = EclUnits.create_unit_system(unit_system)
+
+        return super().pvdx_unit_converter() or ConvertUnits(
+            CreateUnitConverter.ToSI.pressure(unit_system),
+            [
+                CreateUnitConverter.ToSI.recip_fvf(unit_system),
+                CreateUnitConverter.ToSI.compressibility(unit_system),
+                CreateUnitConverter.ToSI.recip_fvf_visc(unit_system),
+                CreateUnitConverter.ToSI.compressibility(unit_system),
+            ],
+        )
+
+    def create_water(
+        self,
+        raw: EclPropertyTableRawData,
+        unit_system: int,
+        surface_mass_densities: List[float],
+    ) -> None:
+        """Creates interpolants for water from the given raw Eclipse data and uses
+        a water unit converter based on the given unit system.
+
+        Args:
+            raw: Eclipse raw data object
+            unit_system: Integer representation of the unit system used in Eclipse data
+
+        """
+        # Holding raw.num_tables values
+        ret: List[PvxOBase] = []
+
+        cvrt = self.water_unit_converter(unit_system)
+
+        ret = self.make_interpolants_from_raw_data(
+            raw,
+            lambda table_index, raw: WaterImpl(table_index, raw, cvrt),
+        )
+
+        assert len(surface_mass_densities) == len(ret)
```

Comment:
Getting `AssertionError` when testing `len(surface_mass_densities) == 2` and `len(ret)==1`.
In general: more descriptive errors than `AssertionError` is of course preferable, but since this is when reading from `INIT` file, it is also a bug 🐛